content/en/docs/plugins/developers-guide.md

```diff
@@ -189,13 +183,25 @@ type Closer interface {
 	Close()
 }
 
+type InitSchema interface {
+	InitSchema() *SchemaInfo
+}
+
+type OpenParams interface {
+	OpenParams() ([]OpenParam, error)
+}
+
 type Progresser interface {
 	Progress(pState sdk.PluginState) (float64, string)
 }
 ```
 
 Source and extrator plugins can optionally implement the `sdk.Destroyer` interface. In that case, `Destroy()` will be called when the plugin gets destroyed and can be used to release any allocated resource.
 
+Source and extrator plugins can also optionally implement the `sdk.InitSchema` interface. In that case, `InitSchema()` will be used to to return a schema describing the data expected to be passed as a configuration during the plugin initialization. This follows the semantics documented for [`get_init_schema`](../plugin-api-reference/#const-char-plugin-get-init-schema-ss-plugin-schema-type-schema-type-required-no). Currently, the schema must follow the [JSON Schema specific](https://json-schema.org/), which in Go can also be easily auto-generated with external packages (e.g. [alecthomas/jsonschema](https:/github.com/alecthomas/jsonschema)).
+
+Source plugins can optionally implement the `sdk.OpenParams` interface. In that case, `OpenParams()` will be called before opening the event stream to obtains some suggested values that would valid parameters for `Open()`. For more details, see the documentation of [`list_open_params`](../plugin-api-reference/#const-char-plugin-list-open-params-ss-plugin-t-s-ss-plugin-rc-rc-required-no).
```

Comment:
```suggestion
Source plugins can optionally implement the `sdk.OpenParams` interface. If requested by the application, the framework may call `OpenParams()` before opening the event stream to obtains some suggested values that would valid parameters for `Open()`. For more details, see the documentation of [`list_open_params`](../plugin-api-reference/#const-char-plugin-list-open-params-ss-plugin-t-s-ss-plugin-rc-rc-required-no).
```

Reply:
Does the framework call `OpenParams()`  always?

Reply:
Same as `InitSchema()`, see the comment above. However, this suggestion is correct because, unlike `get_init_schema`, `list_open_params` is only executed by the framework if requested by the `libs` application.